### PR TITLE
Join discussion after poll test

### PR DESCRIPTION
**Pull request by GHaberis**
## What does this change?

This adds an AB test to check whether showing users a 'join the discussion' message after a poll submission increases likelihood to comment.
## What is the value of this and can you measure success?

Measuring against a baseline of comments on pages with polls that don't show the message after a poll submission.
## Does this affect other platforms - Amp, Apps, etc?

No
## Screenshots

ab-join-discussion-after-poll
## Request for comment

@gtrufitt @guardian/participation @gidsg

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/experiments/join-discussion-after-poll.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+    var showMessage = false;
+
+    function init(isControl) {
+        if (!isControl) {
+            showMessage = true;
+        }
+
+        pollElems = document.querySelectorAll('figure.element-interactive[data-canonical-url*="interactive.guim.co.uk/participation/poll"]');
+
```
Comment (NathanielBennett) [inline static/src/javascripts/projects/common/modules/experiments/join-discussion-after-poll.js]: You totally want to use either jQuery('$') or 'qwery' here. They're both much more efficient

File: static/src/javascripts/projects/common/modules/experiments/join-discussion-after-poll.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+                        showMessageElem(pollElem);
+                    }
+                }
+            }
+        }
+    }
+
+    function matchPostMessageSource(source, pollElem) {
+        var iframe = pollElem.querySelector('iframe');
+
```
Comment (NathanielBennett) [inline static/src/javascripts/projects/common/modules/experiments/join-discussion-after-poll.js]: use '$'

**Comment by gidsg** [comment]
Looks good to me. You might want to squash some of the commits with the same message. 👍 ⏩

File: static/src/javascripts/projects/common/modules/experiments/join-discussion-after-poll.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        return iframe.contentWindow === source;
+    }
+
+    function showMessageElem(pollElem) {
+        var nextSibling = pollElem.nextSibling;
+        var messageElem = document.createElement('div');
```
Comment (NathanielBennett) [inline static/src/javascripts/projects/common/modules/experiments/join-discussion-after-poll.js]: Also, maybe use bonzo for this

**Comment by NathanielBennett** [comment]
LGTM too: :+1:

**Pull request merged**